universal-application-tool-0.0.1/app/repository/VersionRepository.java

```diff
@@ -213,126 +178,40 @@ private BlockDefinition updateQuestionVersions(BlockDefinition block, Transactio
   }
 
-  /** Find an ACTIVE version of the question specified, other than the given ID. */
-  public Optional<Question> findActiveVersionOfQuestion(long questionId, String questionName) {
-    return ebeanServer
-        .find(Question.class)
-        .where()
-        .eq("lifecycle_stage", LifecycleStage.ACTIVE.getValue())
-        .eq("name", questionName)
-        .ne("id", questionId)
-        .findOneOrEmpty();
+  public List<Version> listAllVersions() {
+    return ebeanServer.find(Version.class).findList();
   }
 
-  public void updateProgramsForNewDraftQuestion(
-      Question newDraft, long oldId, Transaction transaction) {
-    // Get all existing draft programs.
-    Stream<Program> draftStream =
-        ebeanServer
-            .find(Program.class)
-            .where()
-            .eq("lifecycle_stage", LifecycleStage.DRAFT)
-            .findList()
-            .stream();
-
-    // Get all programs where there is no draft but where the active version references
-    // this question, and create a draft.
-    Stream<Program> newDraftsStream =
-        ebeanServer
-            .find(Program.class)
-            .alias("programs")
-            .where()
-            .eq("lifecycle_stage", LifecycleStage.ACTIVE)
-            .notExists(
-                ebeanServer
-                    .find(Program.class)
-                    .where()
-                    .raw("programs.name = name")
-                    .eq("lifecycle_stage", LifecycleStage.DRAFT)
-                    .query())
-            .findList()
-            .stream()
-            .filter(
-                program -> {
-                  for (BlockDefinition blockDefinition :
-                      program.getProgramDefinition().blockDefinitions()) {
-                    for (ProgramQuestionDefinition question :
-                        blockDefinition.programQuestionDefinitions()) {
-                      if (question.id() == oldId) {
-                        return true;
-                      }
-                    }
-                  }
-                  return false;
-                })
-            .map(
-                program -> {
-                  ProgramDefinition.Builder newDraftProgram =
-                      program.getProgramDefinition().toBuilder()
-                          .setLifecycleStage(LifecycleStage.DRAFT);
-                  Program programBean = newDraftProgram.build().toProgram();
-                  programBean.setVersion(getNextVersion());
-                  programRepository.insertProgramSync(programBean);
-                  return programBean;
-                });
-
-    Streams.concat(draftStream, newDraftsStream)
-        .forEach(
-            program -> {
-              // Create a draft version of each of those programs.
-              updateQuestionVersions(program, transaction);
-            });
+  public void setLive(long versionId) {
+    try {
+      ebeanServer.beginTransaction();
+      Version draftVersion = getDraftVersion();
```

Comment:
question: when will this method be used? Will we end up generating unnecessary draft versions, since if one does not already exist we create a new draft, only to then mark as DELETED?